TKeazirian.HTTPServer/Router/RoutesConfig.cs

```diff
@@ -1,30 +1,33 @@
+using TKeazirian.HTTPServer.Response;
+
 namespace TKeazirian.HTTPServer.Router;
 
 using Handler;
+using static HttpMethods;
 
-public class RoutesConfig
+public static class RoutesConfig
 {
-    public readonly Dictionary<string, Handler> Routes;
-
-    public RoutesConfig()
+    public static Routes ConstructRoutes()
     {
-        Routes =
-            new Dictionary<string, Handler>
-            {
-                { "/echo_body", new EchoBodyHandler() },
-                { "/simple_get", new SimpleGetHandler() },
-                {
-                    "/simple_get_with_body", new SimpleGetHandler()
-                },
-                { "/redirect", new RedirectHandler() },
-                { "/head_request", new SimpleHeadHandler() },
-                { "/method_options", new SimpleOptionsHandler() },
-                { "/method_options2", new SimpleOptionsHandler2() },
-            };
-    }
+        Routes routes = new();
 
-    public RoutesConfig(Dictionary<string, Handler> routes)
-    {
-        Routes = routes;
+        routes.AddRoute(
+            new("/simple_get", new() { Get }, new SimpleGetHandler())
+        );
+        routes.AddRoute(
+            new("/simple_get_with_body", new() { Get }, new SimpleGetHandler())
+        );
+        routes.AddRoute(
+            new("/echo_body", new() { Post }, new EchoBodyHandler()));
```

Comment:
since you have the path configured here in the dictionary, do you need it in the `Route` instance as well? Do you actually reference `Route.path`?

Reply:
This was changed to be more explicit.  The IDE had suggested removing the types.

Now it looks like this: 

```csharp
new Route("/echo_body", new List<HttpMethod> { HttpMethod.POST }, new EchoBodyHandler()));
```

Reply:
That def reads better! I think what I was getting at though is the the path ends up getting set in two places: as the key in the router dictionary, and also as a property on the `Route` itself. I think it's not particularly problematic, since it seems to only be set in one place. Otherwise my concern would be that you could end up with some discrepancy between the path in the key and the path configured on the route the key points to. But that doesn't seem to be a real issue here, so my real question is, is the path actually doing anything important as a property on the route instance, since it's already represented in the router?

TL;DR: are there actually any dependencies on the path property of the route class? Or is it essentially cruft?

Reply:
I reference `route.Path` in Routes.cs (line 17).

```csharp
_routes.Add(route.Path, route);
```   

That sets the whole Route object, adding a Route to Routes.  As you mentioned, I'm only ever setting the path in the `AddRoute()` method in the RoutesConfig.  But I'm not sure, maybe I don't understand fully what you're saying?

Reply:
Okay okay, I got it now - I understand!  Will update PR soon.